### PR TITLE
WIP: Rewrite most parts of EnergyDispersion class (Fits I/O)

### PR DESCRIPTION
**Pull request by kingj90**
This PR updates the EnergyDispersion class that wraps the OGIP RMF format: 
Changes include:
* Fits output transferred from external np_to_rm function to class member and rewritten (use numpy instead of for loops)
* Energy handling updated in order to use gammapy.spectrum.energy (including some updates in the Energy classes)

NOT included
* Update RMF reader (from_hdu_list)
* Updates of plot routines
* Test for EnergyDispersion2D -> EnergyDispersion

### CODE REVIEW

**Comment by cdeil** [comment]
Let me know if you have any questions or when I should look at something or try it out ...

**Comment by cdeil** [comment]
@kingj90 – I hope #352 won't conflict with this PR!
Apologies ... I did touch the effective area and energy resolution classes again.

**Comment by kingj90** [comment]
@cdeil 
As discussed I wont touch any of the plot routines, only fits/io and energy handling. so there should not be too much interference, I will have a ready to merge version of the IRF classes until tomorrow

**Comment by kingj90** [comment]
This is becoming rather large again, I suggest we leave it for this PR (after adding some tests)

**Comment by cdeil** [comment]
Is this sphinx build error unrelated or was it introduced in this PR?
https://travis-ci.org/gammapy/gammapy/jobs/80646680#L1642
Can you check locally with `python setup.py build_sphinx -l`?

Are you still working on this or should I review now?

File: gammapy/irf/tests/test_energy_dispersion.py [hunk 1]
```diff
@@ -48,3 +62,12 @@ def test_EnergyDispersion2D():
     e_val = (energy + energy2) / 2
     actual = edisp.evaluate(offset, e_val, migra)
     # assert_equal(lower > actual and actual > upper, True)
+
+    # Check RMF exporter
+    offset = Angle([0.612], 'deg')    
+    e_reco = EnergyBounds.equal_log_spacing(1,10,5,'TeV')
+    rmf = edisp.to_energy_dispersion(e_reco, offset)
+
+    actual = rmf._pdf_matrix[5]
```
Comment (cdeil) [inline gammapy/irf/tests/test_energy_dispersion.py]: Ideally tests shouldn't have to access private things like `_pdf_matrix` here.
Maybe it makes sense to expose this as a property or make it a public member (i.e. remove the underscore)?

**Comment by kingj90** [comment]
I am still working on this, since I keep discovering bugs while adding the tests

**Comment by cdeil** [comment]
I think the reason for the travis-ci fail was the matplotlib intersphinx fetch timeout.

After re-starting, travis-ci now shows green light on this build:
https://travis-ci.org/gammapy/gammapy/jobs/80798983

**Comment by cdeil** [comment]
Can you please edit the description of this PR at the top, adding two sentences what's done here and probably removing the task list?

**Comment by kingj90** [comment]
I will clean up a little bit and the ask for comments

**Comment by cdeil** [comment]
I looked over this 5 minutes ago and it looks good.

I don't have time to test this or review this today, so my advice to be either to merge as-is,
or to look at the code coverage for these parts of the code and add tests for things that are completely untested at the moment.

In any case ... feel free to push the "Merge pull request" button yourself when ready!

**Pull request merged**